### PR TITLE
Add clarity for error as the model in route substate

### LINKED ISSUE #1500: Unable to override model hook in error substate route
@f3ndot Reopening your issue emberjs/ember.js/issues/11321 in the guides repo.

> I'm unable to override the model hook in ErrorRoute: the model is null in setupController and any reference to this.get('model') in the ErrorController is also null.

> Potentially adding a sentence to the guides clarifying that you can't override the model hook on sub-states would also be beneficial.

@pixelhandler suggested:
> Perhaps we just need to add to the docs at https://guides.emberjs.com/v2.5.0/routing/loading-and-error-substates/ that the model hooks of an error substate are NOT callled only the setupController method

https://github.com/emberjs/guides/blob/master/source/localizable/routing/loading-and-error-substates.md#error-substates

### PR DESCRIPTION
**Pull request by pixelhandler**
Fixes #1500

Worked with @claubacher to address the issue originally opened by @f3ndot in the ember.js repo.

### CODE REVIEW

**Comment by f3ndot** [comment]
Wonderful, thanks for taking care of this

File: source/localizable/routing/loading-and-error-substates.md [hunk 1]
```diff
@@ -152,6 +152,17 @@ that substate (without updating the URL). The "reason" for the error
 (i.e. the exception thrown or the promise reject value) will be passed
 to that error state as its `model`.
 
+The model hooks (`beforeModel`, `model`, and `afterModel`) of an error substate
+are NOT called. Only the `setupController` method of the error substate is
```
Comment (locks) [inline source/localizable/routing/loading-and-error-substates.md]: downcase "NOT"

File: source/localizable/routing/loading-and-error-substates.md [hunk 2]
```diff
@@ -152,6 +152,17 @@ that substate (without updating the URL). The "reason" for the error
...
+called with the `error` as the model. See example below:
+
+```js
+setupController: function(controller, error) {
+  Ember.Logger.debug(error.message);
+  this._super(controller, error);
```
Comment (locks) [inline source/localizable/routing/loading-and-error-substates.md]: why not `this._super(...arguments)`?

Comment (pixelhandler) [inline source/localizable/routing/loading-and-error-substates.md]: @locks was thinking that using `(controller, error)` helps clarify that it's an error not a model object.

**Comment by pixelhandler** [comment]
@locks I made the changes per your review.

**Comment by locks** [comment]
gracias 😁 
@homu r+

**Pull request merged**